core/templates/dev/head/domain/exploration/SimpleEditorQuestionsDataService.js

```diff
@@ -0,0 +1,150 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for storing information about the current exploration
+ *   in the form of question fields.
+ */
+
+oppia.factory('SimpleEditorQuestionsDataService', [
+  'explorationData', 'explorationInitStateNameService',
+  'explorationStatesService',
+  function(
+      explorationData, explorationInitStateNameService,
+      explorationStatesService) {
+    var questions = null;
+
+    var SUPPORTED_INTERACTION_TYPES = [{
+      id: 'MultipleChoiceInput',
+      name: 'Multiple choice'
+    }];
+
+    var allowedInteractionIds = SUPPORTED_INTERACTION_TYPES.map(
+      function(interactionData) {
+        return interactionData.id;
+      }
+    );
+
+    // Attempts to convert the exploration into a set of fields. If this is
+    // successful, returns the extracted fields, otherwise returns null.
+    var getQuestions = function() {
+      var stateNamesInOrder = [];
+      var currentStateName = explorationInitStateNameService.savedMemento;
+      var questionData = [];
+
+      var iterations = 0;
+      while (currentStateName) {
```

Comment:
I actually quite like the iteration counter, to prevent against the browser locking up unexpectedly. What is your objection?